### PR TITLE
Added label on Data Dictionary, Case List explorer Integration with DD

### PR DESCRIPTION
**Pull request by pxwxnvermx**
## Product Description

This change adds labels for case properties on data dictionaries and integrates the DD labels with case list explorer.

image

## Technical Summary
[Ticket Link](https://dimagi-dev.atlassian.net/browse/INDIV-75)

## Feature Flag
[https://staging.commcarehq.org/hq/flags/edit/data_dictionary/](https://staging.commcarehq.org/hq/flags/edit/data_dictionary/)
[https://staging.commcarehq.org/hq/flags/edit/case_list_explorer/](https://staging.commcarehq.org/hq/flags/edit/case_list_explorer/)

### Safety story

Tested Locally
### Migrations

- [x] The migrations in this code can be safely applied first independently of the code

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/reports/static/reports/js/filters/case_list_explorer.js
Comment (mkangia) [inline corehq/apps/reports/static/reports/js/filters/case_list_explorer.js:78]: :nit leftover?

File: corehq/apps/data_dictionary/models.py
Comment (pxwxnvermx) [reply corehq/apps/data_dictionary/models.py]: d41b5b1ef2f8b348abfe4c5512c6963b13bf9451

File: corehq/apps/data_dictionary/templates/data_dictionary/base.html
Comment (pxwxnvermx) [reply corehq/apps/data_dictionary/templates/data_dictionary/base.html]: i went with the alternative suggestion added the label field below the case_property name
8f868093de4ecf55fa978ed72351f6f03308e47f

**Review by orangejenny — APPROVED** [review]

File: corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js
Comment (mkangia) [inline corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js:33]: how come this one is empty string?

File: corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js
Comment (mkangia) [inline corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js:234]: should this one be `self.newPropertyName()` instead of blank value since by default we are keeping them same or we are not doing that anymore?

File: corehq/apps/data_dictionary/templates/data_dictionary/base.html
Comment (mkangia) [inline corehq/apps/data_dictionary/templates/data_dictionary/base.html:200]: Nice :+1:

File: corehq/apps/data_dictionary/views.py
Comment (mkangia) [inline corehq/apps/data_dictionary/views.py:482]: :facepalm: 
not that you need to change it.

File: corehq/apps/reports/static/reports/js/filters/case_list_explorer.js
Comment (mkangia) [inline corehq/apps/reports/static/reports/js/filters/case_list_explorer.js:59]: what does this section do?

**Review by mkangia — APPROVED** [review]
Confirming that we would update the documentation post merge?

File: corehq/apps/data_dictionary/templates/data_dictionary/base.html
Comment (mkangia) [reply corehq/apps/data_dictionary/templates/data_dictionary/base.html:200]: This needs the screenshots to be updated in the description.

File: corehq/apps/reports/static/reports/js/filters/case_list_explorer.js
Comment (pxwxnvermx) [reply corehq/apps/reports/static/reports/js/filters/case_list_explorer.js:59]: this section updates the label when a user selects a case_property.

File: corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js
Comment (pxwxnvermx) [reply corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js]: thanks for pointing that out. b3ec55464c452eba288843ee58943191d5b57ff4

File: corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js
Comment (pxwxnvermx) [reply corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js:33]: this is a property group so label field is not displayed here, hence the empty string.

File: corehq/apps/data_dictionary/templates/data_dictionary/base.html
Comment (pxwxnvermx) [reply corehq/apps/data_dictionary/templates/data_dictionary/base.html:200]: added the screenshot for updated view.

Comment (mkangia) [reply corehq/apps/data_dictionary/templates/data_dictionary/base.html:200]: Is that most recent? It says "click to add a label" but it should have the same content as case property?
Not sure how this looks now, but you could also see how its done when we add a new question to a form in app builder. The text is hazy which looks nice that it is default and can be changed. That would be a nice cosmetic change, so not essential but good from user experience.

File: corehq/apps/reports/static/reports/js/filters/case_list_explorer.js
Comment (mkangia) [reply corehq/apps/reports/static/reports/js/filters/case_list_explorer.js:59]: Okay. I was confused by the method extraction, `_value`.

File: corehq/apps/data_dictionary/templates/data_dictionary/base.html
Comment (pxwxnvermx) [reply corehq/apps/data_dictionary/templates/data_dictionary/base.html:200]: Yes, that is a recent screenshot, the case_properties were already added, that is why the placeholder is "click to add a label", if a new case_property is added the label is set to the property name.

Comment (mkangia) [reply corehq/apps/data_dictionary/templates/data_dictionary/base.html:200]: Offlined with Pawan that we would reconsider this difference in behaviour for existing and new case properties.

**Comment by mkangia** [comment]
Hey @pxwxnvermx,

The recent commit changes look good. I believe the screenshots in description are yet to be updated for the recent change?

And is this deployed on staging?

**Comment by pxwxnvermx** [comment]
Hi @mkangia, This change is deployed on staging. Updated the screenshots also.

**Comment by mkangia** [comment]
Thanks @pxwxnvermx 

Looking at the screenshots now, I think we would need to reconsider the positioning/placeholder. 
Since the labels don't have their own headers now, its not clear what the input box is for, so we would need to either give it, its own column, or keep placeholder as "Click to add label".
I think the former would make sense, because if a custom label is added it's not clear what it is and can be confusing for someone looking at DD for the first time.
Or if we could add a small label (HTML label) that says "Label" before the input box for labels.
@orangejenny if this makes sense, any suggestions?

**Comment by orangejenny** [comment]
@mkangia Yeah, I agree with you, it's not quite clear what the labels are. I like the suggestion of "Click to add label" as placeholder.

Also FYI @biyeun in case you have a suggestion.

**Comment by pxwxnvermx** [comment]
Thanks @mkangia @orangejenny for the inputs.

I had one more concern regarding this, changing the placeholder to "Click to add a label" will work when the label field is empty. 
What can we do for the cases where the label field is filled with the property name itself, as it is the default label and will get saved once the user presses the save button, in that case how can we indicate to the user that the pre-filled input here is the Label for the above case property?

image

As Manish said we can use a small label (html) `Label:` to indicate that this input is the Label.

**Comment by mkangia** [comment]
Hey @pxwxnvermx 
Since you have access to the current changes good to try the following approaches or any new that you can think of during redesigning
1. add a small label "Label" along with having the default value already filled
2. adding a new column itself for labels with having the default value already filled
3. any other that you can think of

**Comment by orangejenny** [comment]
Forgot about the default values. Agree that a small "label" label could work. You could also move the label into the description column & update the header there to "Label & Description"

**Comment by biyeun** [comment]
> @mkangia Yeah, I agree with you, it's not quite clear what the labels are. I like the suggestion of "Click to add label" as placeholder.
> 
> Also FYI @biyeun in case you have a suggestion.

👍 yes a placeholder is a great first step.

Is the word "label" clear enough to the user as to what it actually does? Is there documentation associated with this change to clarify any questions the user might have about this? For instance... What is the difference between "label" and "description"? Where does "label" appear when the data dictionary is referenced? How is it displayed to other users?

**Comment by esoergel** [comment]
Out of scope of this PR, but I just wanted to note that this field will be a nice addition to the `Data Dictionary Case Data Page` feature flag (https://github.com/dimagi/commcare-hq/pull/32158).  I'll probably pull it in there some time after this is merged.

**Comment by mkangia** [comment]
Sounds great @esoergel.

**Comment by mkangia** [comment]
Documentation completed. Just waiting on final confirmation to merge this.

**Pull request merged**
